packages/near-membrane-base/src/membrane.ts

```diff
@@ -1320,11 +1400,12 @@ export function createMembraneMarshall() {
                 lastProxyTrapCalled = ProxyHandlerTraps.PreventExtensions;
                 if (ReflectIsExtensible(shadowTarget)) {
                     const { foreignTargetPointer } = this;
-                    if (!foreignCallablePreventExtensions(foreignTargetPointer)) {
-                        // if the target is a proxy manually created, it might
-                        // reject the preventExtension call, in which case we
-                        // should not attempt to lock down the shadow target.
-                        if (!foreignCallableIsExtensible(foreignTargetPointer)) {
+                    const result = foreignCallablePreventExtensions(foreignTargetPointer);
+                    if (result & PreventExtensionsResult.False) {
+                        if (!(result & PreventExtensionsResult.Extensible)) {
+                            // If the target is a proxy manually created, it might
+                            // reject the preventExtension call, in which case we
+                            // should not attempt to lock down the shadow target.
```

Comment:
☝️ Batched `foreignCallablePreventExtensions` and `foreignCallableIsExtensible`.